js/serial_manager.js

```diff
@@ -0,0 +1,590 @@
+		GUI.operating_mode = 1; // we are connected
+		$('#tabs li.tab_TX a').click(); // open TX tab
+	}
+	// load_tx_tab();
+	PSP.send_message(PSP.PSP_REQ_TX_CONFIG, false, false, get_active_profile);
+};
+serial_manager.prototype.startup_cb = function() {
+	// change connect/disconnect button 
+	// from "connecting" status to disconnect
+	this.port_picker.addClass('active').text(chrome.i18n.getMessage('disconnect'));
+
+	if (!initialize_configuration_objects(CONFIGURATOR.firmwareVersionLive)) {
+		this.fail('fw_not_supported');
+		return;
+	}
+	if (GUI.module == 'TX') {
+		this.init_tx_tab();
+	} else {
+		this.init_rx_tab();
+	}
+};
+serial_manager.prototype.connect_std = function() {
+	var that = this;
+	this.startup_message_buffer = '';
+	GUI.timeout_add('startup', function () {
+		// reset the connect button back to "disconnected" state
+		that.port_picker.click(); 
+		GUI.log(chrome.i18n.getMessage('error_no_startup_message'));
+	}, 10000);
+	this.ref = this.startup_listener.bind(this);
+	serial.onReceive.addListener(this.ref);
+};
+serial_manager.prototype.connect_quick = function() {
+	var that = this;
+   // quick join (for modules that are already in bind mode 
+   // and modules connected through bluetooth)
+	this.ref = this.read_serial.bind(this);
+	serial.onReceive.addListener(this.ref);
+
+	// using this timeout as protection against locked bus 
+	// (most likely chrome serial api bug), if sending "B" fails
+	// PSP callback with timeout trigger wouldn't trigger
+	GUI.timeout_add('send_timeout', function () {
+		GUI.log(chrome.i18n.getMessage('error_failed_to_enter_binary_mode'));
+
+		// disconnect
+		that.port_picker.click();
+	}, 250);
+
+	// B char == enter binary mode
+	this.send("B", function() {
+		PSP.send_message(PSP.PSP_REQ_FW_VERSION, false, false, function (result) {
+			GUI.timeout_remove('send_timeout');
+			if (result) {
+				console.log('Quick join success');
+				GUI.connected_to = GUI.connecting_to;
+				GUI.connecting_to = false;
+				chrome.storage.local.set({'last_used_port': GUI.connected_to});
+				that.connect_quick_cb();
+			} else {
+				console.log('Quick join expired');
+				// standard connect sequence uses its own listener
+				serial.onReceive.removeListener(that.ref);
+				// continue
+				that.connect_wrapper();
+			}
+		}, 200);
+	});
+};
+serial_manager.prototype.connect_quick_cb = function() {
+	var that = this;
+	PSP.send_message(PSP.PSP_REQ_TX_CONFIG, false, false, function (result) {
+		if(result){
+			GUI.module = 'TX';
+		} else {
+			GUI.module = 'RX';
+		}
+		that.startup_cb();
+	});
+};
+serial_manager.prototype.connect_wrapper = function(){
+	if (GUI.optional_usb_permissions) {
+		// check if 32u4 is present
+		this.check_usb_devices();
+	} else {	
+		this.connect_std();
+	}
+};
+
+serial_manager.prototype.check_usb_devices = function () {
+	var that = this;//
+	chrome.serial.getDevices(function(dev){
+		if(!dev || !dev.length) {console.log('error');that.close();}
```

Comment:
Missing newlines.

Reply:
oops, forgot to clean that up... it was meant just for testing. will fix...